### DISCUSSION TITLE
Error creating the new SMP Service Group for participant

### DISCUSSION THREAD

**Discussion by ThodorisGiouroukos** [phax/phoss-smp#280] [General]
Hi, 

After updating the certificates in both Test and Production we receive the following error when we try to add a new participant: 

Error creating the new SMP Service Group for participant 'iso6523-actorid-upis::XXXX:XXXXXXXXX'.
Technical details: SMPSMLException - Failed to create 'iso6523-actorid-upis::XXXX:XXXXXXXXX' in SML
Caused by: RegistrationHookException - Could not create business iso6523-actorid-upis::XXXX:XXXXXXXXX in SML
Caused by: ClientTransportException - The server sent HTTP status code 401: Unauthorized

The smp-keystore has 3 certificates in the chain. 
1. PEPPOL Root TEST CA - G2
2. PEPPOL SERVICE METADATA PUBLISHER TEST CA - G2
3. PGR000xxx

When trying to enter the New public certificate in the SML certificate update section, we get the error:

Error preparing migration of SMP certificate at SML 'https://acc.edelivery.tech.ec.europa.eu/edelivery-sml'.
Technical details: ClientTransportException - The server sent HTTP status code 401: Unauthorized

What are your suggestions ?

Thanks,
Thodoris

**Comment by phax** [comment]
Hello @ThodorisGiouroukos,

The HTTP status code 401 on Service Group creation usually has one of the following two reasons:
1. The participant is already registered at a different Service Provider. Please use https://peppol.helger.com/public/locale-en_US/menuitem-tools-pid-check to check, if that participant is already registered.
2. You recently did an update of your SMP certificate, and you forgot to inform the SML about your new SMP certificate. Please see https://peppol.helger.com/public/locale-en_US/menuitem-docs-peppol-cert-update#smp for the process to update it.

hth, Philip

**Reply by ThodorisGiouroukos** [reply]
Hi Philip,

Thank you for your reply!
We have double check that the participant is not already registered, so it must be the 2nd reason.
As I mentioned earlier, we did a SMP certificate update recently. I've tried already to follow the process to update the SML certificate but I get:
"Error preparing migration of SMP certificate at SML 'https://acc.edelivery.tech.ec.europa.eu/edelivery-sml'.
Technical details: ClientTransportException - The server sent HTTP status code 401: Unauthorized"

![Screenshot 2024-06-27 113810](https://github.com/phax/phoss-smp/assets/173901712/63417cb4-88f8-48cd-b59c-5c0eff32e3c4)

**Reply by phax** [reply]
Okay, then you need to read the content of the link more closely ;-)
> If your certificate is already expired, you need to contact [EC eDelivery support](https://peppol.helger.com/public/menuitem-docs-sml-support) and provide the following information: old certificate subject and certificate serial number as well as the new certificate file (PEM encoded, should start with -----BEGIN CERTIFICATE----- and end with -----END CERTIFICATE-----).
Note: you may already update your SMP software with the new certificate, because the old certificate is anyway useless.
Note: no further action or service call to the SML is necessary, if you are using this manual process.

**Comment by ThodorisGiouroukos** [comment]
The SMP certificate has already been renewed and it seems to work fine. The cert was renewed using the manual way, with the keystore.
Should we follow the process you advised?

**Reply by phax** [reply]
I'd say we are finished :) You updated your certificates, you can create new Service Groups - problem solved. Or am I missing something?
